BaragonServiceIntegrationTests/src/test/java/com/hubspot/baragon/BaragonServiceTestIT.java

```diff
@@ -0,0 +1,478 @@
+    baragonServiceClient.enqueueRequest(request);
+  }
+  
+  private void removeService(String serviceId) {
+    System.out.println("Cleaning up...");
+    BaragonResponse resp = baragonServiceClient.deleteService(serviceId).get();
+    while(baragonServiceClient.getRequest(resp.getLoadBalancerRequestId()).get().getLoadBalancerState().equals(BaragonRequestState.WAITING)) { }
+    
+    assertFalse(baragonServiceClient.getServiceState(serviceId).isPresent());
+  }
+  
+  // ----------------------------------------------------
+  
+  @Before
+  public void setup() throws Exception {
+    baragonServiceClient = getInjector().getInstance(BaragonServiceClient.class);
+  }
+  
+  @Rule
+  public TestRule watcher = new TestWatcher() {
+     protected void starting(Description description) {
+        System.out.println("Starting: " + description.getMethodName());
+     }
+     protected void succeeded(Description description) {
+       System.out.println("\u001B[32mTest passed\u001B[0m");
+     }
+     protected void failed(Throwable e, Description description) {
+       System.out.println("\u001B[31mTest failed\u001B[0m");
+     }
+  };
+  
+  
+  @Rule
+  public Stopwatch stopwatch = new Stopwatch() {
+      @Override
+      protected void finished(long nanos, Description description) {
+        System.out.println("(" + nanos / 1000000000.0 + " s)");
+      }
+  };
+  
+  @After
+  public void teardown() {
+    baragonServiceClient = null;
+  }
+
+  // ----------------------------------------------------
+  
+  @Test
+  public void testStatus() throws Exception {
+    Optional<BaragonServiceStatus> status = baragonServiceClient.getAnyBaragonServiceStatus();
+    assertTrue(status.get().isLeader());
+  }
+
+  @Test
+  public void testWorkers() throws Exception {
+    assertFalse(baragonServiceClient.getBaragonServiceWorkers().isEmpty());
+  }
+  
+  @Test
+  public void testLoadBalancers() throws Exception {
+    assertFalse(baragonServiceClient.getLoadBalancerGroups().isEmpty());
+  }
+  
+  @Test
+  public void testClusterAgents() throws Exception {
+    BaragonAgentMetadata metadata = baragonServiceClient.getLoadBalancerGroupAgentMetadata(LOAD_BALANCER_GROUP).toArray(new BaragonAgentMetadata[0])[0];
+    assertTrue(metadata.getBaseAgentUri() != null && !metadata.getBaseAgentUri().isEmpty());
+  }
+  
+  @Test
+  public void testClusterKnownAgents() throws Exception {
+    BaragonAgentMetadata metadata = baragonServiceClient.getLoadBalancerGroupKnownAgentMetadata(LOAD_BALANCER_GROUP).toArray(new BaragonAgentMetadata[0])[0];
+    assertTrue(metadata.getBaseAgentUri() != null && !metadata.getBaseAgentUri().isEmpty());
+    
+    testDeleteKnownAgent();
+  }
+  
+  // ----------------------------------------------------
+  
+  @Test
+  public void testValidRequest() throws Exception {
+    String serviceId = setupTestService();
+    while(baragonServiceClient.getRequest(serviceId).get().getLoadBalancerState().equals(BaragonRequestState.WAITING)) { }
```

Comment:
I think it would be good to make these while checks into their own method, maybe using something like [guava-retry](https://github.com/rholder/guava-retrying). It should help us to not get stuck in the case where a request never exists waiting status as well as adding some pauses instead of hitting the endpoint in BaragonService as fast as we can.

e.g.
```java
public void watchForStatus(String requestId, BaragonRequestState targetState) {
  // watch for request to enter targetState and throw an error if it doesn't get to the desired state
}